frontend/src/scenes/error-tracking/errorTrackingLogic.ts

```diff
@@ -44,5 +54,58 @@ export const errorTrackingLogic = kea<errorTrackingLogicType>([
                 setFilterTestAccounts: (_, { filterTestAccounts }) => filterTestAccounts,
             },
         ],
+        sparklineSelectedPeriod: [
+            lastDay.value as string | null,
+            { persist: true },
+            {
+                setSparklineSelectedPeriod: (_, { period }) => period,
+            },
+        ],
+        sparklineOptions: [
+            [lastDay, lastHour] as SparklineOption[],
+            { persist: true },
+            {
+                _setSparklineOptions: (_, { options }) => options,
+            },
+        ],
     }),
+    listeners(({ values, actions }) => ({
+        setDateRange: ({ dateRange: { date_from } }) => {
+            const options: SparklineOption[] = []
+
+            if (date_from === 'dStart' || date_from === '-24h') {
+                // today and last 24 hours
+                options.push(lastDay, lastHour)
+            } else if (date_from === '-1dStart') {
+                // yesterday
+                options.push({ value: '-1d24h', label: '24h' }, { value: '-1d1h', label: '1h' })
+            } else if (date_from === 'mStart') {
+                // this month
+                options.push(lastMonth, lastDay)
+            } else if (date_from === 'yStart') {
+                // this year
+                options.push(lastYear, lastMonth)
+            } else if (date_from === 'all') {
+                // all time
+                options.push(lastYear, lastMonth, lastDay)
+            } else if (date_from) {
+                const isRelative = date_from.match(/-\d+[hdmy]/)
+                if (isRelative) {
+                    const value = date_from?.replace('-', '')
+                    options.push({ value: value, label: value }, lastDay)
+                }
+            }
```

Comment:
the `options.push` confused me here...

since there's all else if (apart from that pesky match) this could be a lookup on a map.. I bet it still could

```suggestion
            // could be defined up top in the file, doesn't need recreating
            const optionsMapping = {
                'dStart': [lastDay, lastHour],
                '-24h':  [lastDay, lastHour],
                '-1dStart': [{ value: '-1d24h', label: '24h' }, { value: '-1d1h', label: '1h' }],
                'mStart': [lastMonth, lastDay],
                'yStart': [lastYear, lastMonth],
                'all': [lastYear, lastMonth, lastDay],
                
            }
           const options: SparklineOption[]  = optionsMapping[date_from]
           if (date_from) {
                const isRelative = date_from.match(/-\d+[hdmy]/)
                if (isRelative) {
                    const value = date_from?.replace('-', '')
                    // TODO does this add or replace?
                    options.push({ value: value, label: value }, lastDay)
                }
            }
```

only a nit